### ISSUE TITLE
Rust buffers do not show line numbers

### ISSUE THREAD

**Issue by rakanalh** [hlissner/doom-emacs#3286]
**What did you expect to happen?**

When a rust file is opened, line numbers are displayed

Calling `M-x line-number-mode` confirms enabled / disabled but line numbers aren't actually displayed.

**What actually happened?**

line numbers are not displayed... tested python files and line numbers are displaying as usual. 

**Steps to reproduce**:
- Open a couple of rust files in splits
- `SPC q r`
- Notice how line numbers are not displayed for automatically loaded buffers.

**Additional details:**
- Include a link to your private config
https://github.com/rakanalh/dotfiles/tree/master/.doom.d

- Include screenshots/casts of your issue
![Screen Capture_select-area_20200603143750](https://user-images.githubusercontent.com/195829/83637951-acc67f00-a5a8-11ea-96a8-111e47f3a9c1.png)

**System information:**
<details><pre>
<details>

```
emacs   version    28.0.50
        features   XPM JPEG TIFF GIF PNG RSVG CAIRO SOUND GPM DBUS GLIB NOTIFY INOTIFY ACL GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD JSON PDUMPER LCMS2 GMP
        build      May 31, 2020
        buildopts  (--prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --localstatedir=/var --mandir=/usr/share/man --with-gameuser=:games --with-sound=alsa --with-modules --without-gconf --without-gsettings --with-x-toolkit=gtk3 --without-xaw3d --without-compress-install 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -flto' CPPFLAGS=-D_FORTIFY_SOURCE=2 LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now)
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop 87364138d 2020-06-03 03:46:58 -0400
        dir        ~/dotfiles/.doom.d/
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /usr/bin/zsh
        uname      Linux 5.6.15-arch1-1 #1 SMP PREEMPT Wed, 27 May 2020 23:42:26 +0000 x86_64
        path       (~/.emacs.d/bin/../bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin /usr/local/git/bin /usr/local/go/bin /opt/sm/bin /opt/sm/pkg/active/bin /opt/sm/pkg/active/sbin /bin ~/Code/Go/bin ~/.bin ~/.cargo/bin /usr/lib/emacs/28.0.50/x86_64-pc-linux-gnu/)
config  envfile    envvar-file
        elc-files  0
        modules    (:completion company ivy :ui doom doom-dashboard doom-quit hl-todo hydra modeline nav-flash ophints (popup +all +defaults) treemacs vc-gutter vi-tilde-fringe window-select workspaces zen :editor (evil +everywhere) file-templates fold (format +onsave) multiple-cursors rotate-text snippets :emacs dired electric ibuffer vc (undo +tree) :term vterm :checkers syntax :tools editorconfig (eval +overlay) (lookup +docsets) (lsp +peek) magit :lang data emacs-lisp go javascript lua markdown (org +dragndrop +ipython +pandoc +present) python (rust +lsp) sh solidity yaml :app irc :config (default +bindings +smartparens))
        packages   ((auto-highlight-symbol) (rust-mode :disable t) (format-all :disable t) (whitespace :disable t) (cargo) (flycheck-mypy) (parent-mode) (rust-playground) (org-journal) (org-present) (protobuf-mode) (google-translate))
        unpin      (rustic lsp-mode lsp-ui)
        elpa       (protobuf-mode)
```
</details>
</pre></details>

**Comment by rakanalh** [comment]
When restarting doom, buffers loaded automatically don't seem to show line numbers. Updated steps to reproduce.

**Comment by unit-374** [comment]
I have the same problem after https://github.com/hlissner/doom-emacs/commit/134554dd69d9b1cea3d2190422de580fddf40ecd.
In `*Messages*`
`File mode specification error: (error nil is not a valid Flycheck syntax checker)`
It causes other modes disfunction (e.g. yas-snippet, line-number-mode).

**Comment by hlissner** [comment]
> File mode specification error: (error nil is not a valid Flycheck syntax checker)

This is likely the cause for OP's issue, which should be fixed as of db25224. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!

**Comment by mattsawyer77** [comment]
appears to be fixed. thanks @hlissner for the fast response!!
